Sledge.Packages/Wad/WadPackage.cs

```diff
@@ -11,23 +12,26 @@ namespace Sledge.Packages.Wad
     // https://developer.valvesoftware.com/wiki/WAD
     public class WadPackage : IPackage
     {
-        private const string Signature = "WAD3";
+        private const string WAD2Signature = "WAD2";
+        private const string WAD3Signature = "WAD3";
 
         public FileInfo PackageFile { get; private set; }
         internal uint NumTextures { get; private set; }
         internal uint LumpOffset { get; private set; }
         internal List<WadEntry> Entries { get; private set; }
+        public Palette Palette { get; private set; }
 
-        public WadPackage(FileInfo packageFile)
+        public WadPackage(FileInfo packageFile, Palette pal)
```

Comment:
I agree that passing the Palette everywhere is messy. The problem with just using `packageFile` is, AFAIK, it's not necessarily inside the game directory - you could add a WAD file from a random location on your hard drive, but the palette always resides in the game directory. To work around that, I think the IFile for the game root directory would also need to be passed to the WadPackage constructor like:

`public WadPackage(FileInfo packageFile, IFile gameRoot)`

then I could do this:
```
if (sig == WAD2Signature) {
    var palette = gameRoot.TraversePath("/gfx/palette.lmp");
    // Load palette byte data
}
```
However, the problem with that is I can't use IFile or TraversePath inside the WadPackage implementation because it creates a circular dependency between Sledge.Filesystem and Sledge.Packages.
Maybe that could be worked around by merging the FileSystem and Packages assemblies, but that seems like a heavy handed workaround.